doc/manual/variables-and-scoping.rst

```diff
@@ -194,28 +223,33 @@ are some extra rules for :ref:`let-blocks <man-let-blocks>` and for
 
 In the following example the ``x`` and ``y`` refer always to the same
 variables as the soft local scope inherits both read and write
-variables::
+variables:
+
+.. doctest::
 
-    x,y = 0, 1
-    for i = 1:10
-        x = i + y + 1
-    end
+    julia> x, y = 0, 1;
+
+    julia> for i = 1:10
+               x = i + y + 1
+           end
 
     julia> x
     12
 
 Within soft scopes, the `global` keyword is never necessary, although
 allowed.  The only case when it would change the semantics is
-(currently) a syntax error::
+(currently) a syntax error:
 
-    let
-        local x = 2
-        let
-            global x = 3
-        end
-    end
+.. doctest::
 
-    # ERROR: syntax: `global x`: x is local variable in the enclosing scope
+    julia> let
+               local x = 2
+               let
+                   global x = 3
+               end
+           end
+    ERROR: syntax: `global #3#x`: #3#x is local variable in the enclosing scope
```

Comment:
Are you referring to the confusion added by `#3#`? I changed the variable used here from `x` to `j` so that the error message returned is cleaner -- ```ERROR: syntax: `global j`: j is local variable in the enclosing scope```